authlib/jose/rfc7516/jwe.py

```diff
@@ -139,6 +141,145 @@ def serialize_compact(self, protected, payload, key, sender_key=None):
             urlsafe_b64encode(tag)
         ])
 
+    def serialize_json(self, header_obj, payload, keys, sender_key=None):
+        if isinstance(keys, tuple):  # tuple instead of list
+            keys = list(keys)
+        if not isinstance(keys, list):  # single key
+            keys = [keys]
+
+        if not keys:
+            raise ValueError("No keys have been provided")
+
+        shared_header = JWESharedHeader.from_dict(header_obj)
+
+        recipients = header_obj.get('recipients')
+        if recipients is None:
+            recipients = [{} for _ in keys]
+        for i in range(len(recipients)):
+            if recipients[i] is None:
+                recipients[i] = {}
+            if recipients[i].get('header') is None:
+                recipients[i]['header'] = {}
+
+        jwe_aad = header_obj.get('aad')
+
+        if len(keys) != len(recipients):
+            raise ValueError("Count of recipient keys {} does not equal to count of recipients {}"
+                             .format(len(keys), len(recipients)))
+
+        # step 1: Prepare algorithms & key
+        alg = self.get_header_alg(shared_header)
+        enc = self.get_header_enc(shared_header)
+        zip_alg = self.get_header_zip(shared_header)
+
+        self._validate_sender_key(sender_key, alg)
```

Comment:
Do we validate recipient keys and check that sender and recipient keys are on the same curve?